src/compile/axis/config.ts

```diff
@@ -1,20 +1,56 @@
-import {ScaleType} from 'vega-typings/types';
-import {array} from 'vega-util';
+import {ScaleType, SignalRef} from 'vega-typings/types';
+import {array, stringValue} from 'vega-util';
 import {AxisConfig} from '../../axis';
 import {PositionScaleChannel} from '../../channel';
 import {Config} from '../../config';
 import {isQuantitative} from '../../scale';
-import {titlecase} from '../../util';
+import {keys, titlecase} from '../../util';
+import {isSignalRef} from '../../vega.schema';
 import {getStyleConfig} from '../common';
 
-function getAxisConfigFromConfigTypes(configTypes: string[], config: Config) {
+function getAxisConfigFromConfigTypes(
+  configTypes: string[],
+  config: Config,
+  channel: 'x' | 'y',
+  orient: string | SignalRef
+) {
   // TODO: add special casing to add conditional value based on orient signal
-  return Object.assign.apply(null, [{}, ...configTypes.map(configType => config[configType])]);
+  return Object.assign.apply(null, [
+    {},
+    ...configTypes.map(configType => {
```

Comment:
This map is getting pretty big. Let's pull it out and use a normal for loop.

Reply:
Why would using loop be better?

Using map and then `assign`ing once also means the merging is done in native code, not JS and thus faster.

Reply:
A loop is easier to step into when using a debugger.